### PR TITLE
adding chapter 7 exercises

### PR DESCRIPTION
**Pull request by MarkWignall**

### CODE REVIEW

File: chapter07/mdw/Fred.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Mr. Slate had an employee
+Fred Filntstone was his name
+fred was happy
+fred was married to wilma
+wilma and betty were friends
+fred was the father of bammbamm
```
Comment (codavex) [inline chapter07/mdw/Fred.txt:6]: Wait, what? Wow. That adds whole new layers to the show. Did Wilma know? Did Barney? Wow. :scream:

File: chapter07/mdw/Fred.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+Alfred yelled for BATMAN.
+barney yelled for FRED
+nobody was in this line
+frederick was alone
+so was sam
+and pat
+writer has A breakdown
+wilma is fred?
```
Comment (codavex) [inline chapter07/mdw/Fred.txt:14]: Seriously, this file needs a spoiler warning or something

File: chapter07/mdw/Fred.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+fred was married to wilma
+wilma and betty were friends
+fred was the father of bammbamm
+Alfred yelled for BATMAN.
+barney yelled for FRED
+nobody was in this line
+frederick was alone
+so was sam
+and pat
+writer has A breakdown
```
Comment (thatguy59) [inline chapter07/mdw/Fred.txt:13]: We noticed.

File: chapter07/mdw/Ex04.pl [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+use strict;
+use warnings;
+use 5.010;
+
+foreach (<>) {
+  if ( /\b[^\s\\]*[A-Z]+[^\s\\]*[a-z]+[^\s\\]*\b/ || /\b[^\s\\]*[a-z]+[^\s\\]*[A-Z]+[^\s\\]*\b/ ) {
+ # comment as i have made this more complicated than needed, this will match any word with anything+capital+anything+non-capital+anything or the othe way around
+    print $_
+  }
```
Comment (JPeace2) [inline chapter07/mdw/Ex04.pl:11]: This is very complicated and is very difficult to understand. I think the easiest solution to this exercise is to do something similar to exercises 6 i.e. if ( /[a-z]/ && /[A-Z]/ ).

Comment (MarkWignall) [inline chapter07/mdw/Ex04.pl:11]: That does not do the same thing. This checks for a single word with both a capital and a non-capital, as opposed to checking the line. I was looking to capture each word and then run that command originally but could not figure out how. I considered splitting the input but felt that was not using regex.

Comment (JPeace2) [inline chapter07/mdw/Ex04.pl:11]: The official solution in the book is /[A-Z][a-z]+/ which I think will match any capital that isn't at the end of the line or before a lower case letter.

File: chapter07/mdw/Ex01.pl [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use 5.010;
+
```
Comment (pratyp) [inline chapter07/mdw/Ex01.pl:6]: <code>strict warnings</code> seem to have worked!

File: chapter07/mdw/Ex06.pl [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use 5.010;
+
+foreach (<>) {
+  if ( /fred/ ) {
+    if ( /wilma/ ) {
+    print $_
```
Comment (pratyp) [inline chapter07/mdw/Ex06.pl:10]: can use <code>&&</code> instead of multiple <code>if</code>'s.

File: chapter07/mdw/Fred.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+wilma is fred?
+freD?
+fRedom!!!
+FLAGRAGa!
+FLLLlRRRG
```
Comment (pratyp) [inline chapter07/mdw/Fred.txt:18]: _**flushing out what she just read and restoring her childhood memories**_  :expressionless:

File: chapter07/mdw/Ex04.pl (hunk 4 above)
Comment (qianliok) [inline chapter07/mdw/Ex04.pl:11]: Mark wants to also match words like IBM, A-Team, etc. :smile:

File: chapter07/mdw/Ex05.pl [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use 5.010;
+
+foreach (<>) {
+  if ( /(\S)\1/ ) {
```
Comment (qianliok) [inline chapter07/mdw/Ex05.pl:8]: Good!

File: chapter07/mdw/Ex04.pl (hunk 4 above)
Comment (MarkWignall) [inline chapter07/mdw/Ex04.pl:11]: This doesn't match IBM because that is all capitals but it does match Sørina, Mátyás and Mónica.

Comment (qianliok) [inline chapter07/mdw/Ex04.pl:11]: hmm... Will `[A-Z]+` not matching `[A-Z]` at least one or more times?

Comment (MarkWignall) [inline chapter07/mdw/Ex04.pl:11]: Yeah, I think I could have removed all the +'s.

**Pull request merged**